src/pipx/package_specifier.py

```diff
@@ -161,7 +161,24 @@ def parse_specifier_for_install(package_spec: str, pip_args: List[str]) -> Tuple
         )
         pip_args.remove("--editable")
 
-    return (package_or_url, pip_args)
+    pip_args = [option for pip_arg in pip_args for option in pip_arg.split(" ")]
+
+    for option in pip_args:
+        if "-c" in option or "--constraint" in option:
+            option_list = option.split("=")
+
+            if len(option_list) != 2:
+                continue
+
+            key, value = option_list
+            value_path = Path(value)
+            if value_path.exists():
```

Comment:
We don't need to perform this check ourselves, `pip` will catch a non-existing constraint file for us.

Reply:
right, changed both lines to `value_path = Path(value).expanduser().resolve()`